contracts/metatx/MinimalForwarder.sol

```diff
@@ -26,29 +21,64 @@ contract MinimalForwarder is EIP712 {
         bytes data;
     }
 
-    bytes32 private constant _TYPEHASH =
+    struct ExpiringForwardRequest {
+        uint48 deadline;
+        address from;
+        address to;
+        uint256 value;
+        uint256 gas;
+        uint256 nonce;
+        bytes data;
+    }
+
+    bytes32 private constant _FORWARD_REQUEST_TYPEHASH =
         keccak256("ForwardRequest(address from,address to,uint256 value,uint256 gas,uint256 nonce,bytes data)");
+    bytes32 private constant _EXPIRING_FORWARD_REQUEST_TYPEHASH =
+        keccak256(
+            "ExpiringForwardRequest(uint48 deadline,address from,address to,uint256 value,uint256 gas,uint256 nonce,bytes data)"
```

Comment:
why put the deadline as the first argument? I would personnaly put it after the nonce.

Reply:
Made sense to me to pack them this way. It's not a bid deal since it's calldata, but I thought it was better to put the `uint48` next to an `address`

Reply:
I believe ABI encoding does not pack values. Every value takes up 32 bytes.